features/step_definitions/metering.rb

```diff
@@ -302,12 +302,24 @@
   https_opts[:headers][:content_type] ||= "application/json"
   https_opts[:headers][:authorization] ||= "Bearer #{bearer_token}"
   # first we need to expose reporting API route if not route is found
-  step %Q/I enable route for metering service/
+  step %Q/I enable route for metering service/ unless route('metering').exists?
   report_name = opts[:report_name]
-  url_path = "/api/v1/reports/get?name=#{report_name}&namespace=#{cb.metering_namespace.name}&format=json"
+  url_path ||= opts[:custom_url]
+  url_path ||= "/api/v1/reports/get?name=#{report_name}&namespace=#{cb.metering_namespace.name}&format=json"
+
   report_query_url = route.dns + url_path
   @result = BushSlicer::Http.request(url: report_query_url, **https_opts, method: 'GET')
   if @result[:success]
     @result[:parsed] = YAML.load(@result[:response])
   end
 end
+Given /^I wait(?: up to ([0-9]+) seconds)? for metering route to be accessible$/ do | seconds |
+  seconds = seconds.to_i unless seconds.nil?
+  seconds ||= 60
+  wait_for(seconds) {
+    step %Q/I perform the GET metering rest request with:/, table(%{
+      | custom_url | /healthy |
+    })
+    @result[:success]
+  }
```

Comment:
there is nothing to make this step fail on timeout. Or do you expect next steps to fail? It is fine as long as error message clearly indicates the route failed to become accessible in time.

Reply:
@akostadinov I added this step to make sure the route is accessible before doing REST query.  Do you want me to add an error message within this step if `@result[:success]` is false?

Reply:
If you want step to fail, then you need to raise an exception. Presently it will wait but on timeout it will just continue without warning.

Reply:
@akostadinov ok, raising error instead of relying on the query step to fail.  PTAL.  Thanks!